src/main/java/jenkins/plugins/openstack/compute/JCloudsCleanupThread.java

```diff
@@ -176,6 +179,9 @@ private void deleteComputer(JCloudsComputer comp, CauseOfInterruption coi) {
     private @Nonnull HashMap<JCloudsCloud, List<Server>> destroyServersOutOfScope() {
         HashMap<JCloudsCloud, List<Server>> runningServers = new HashMap<>();
         for (JCloudsCloud jc : JCloudsCloud.getClouds()) {
+            if (jc.getDisabled().isDisabled()) {
+                continue; // skip any cloud that's disabled.
```

Comment:
FYI the `isDisabled()` method returns true if a cloud is temporarily disabled due to provisioning issues _or_ has been disabled by the user (un-ticking the enabled checkbox).
I think we'd want the user's choice of disabling a cloud to prevent _any_ traffic to that cloud, including cleanup, but I agree that a temporary disablement _might_ be viewed as an insufficient reason to skip cleanup.
...but it is temporary, so the _next_ scheduled cleanup would likely clean up as normal
...and a temporary disable is normally on a specific template rather than the cloud as a whole, and this check is for the cloud as a whole.
So ... yes, we chould change the code here, but I don't see much "real world" benefit in doing so in this case.